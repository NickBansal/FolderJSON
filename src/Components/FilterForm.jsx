import React from 'react'
import '../Stylesheets/FilterForm.css'

const FilterForm = ({ handleSubmit }) => {
    return (
        <div>
            <form onSubmit={(e) => {
                e.preventDefault()
                handleSubmit(e.target[0].value)}}>
                <input type='text' placeholder='Filter...' />
            </form>
        </div>
    )
}

export default FilterForm