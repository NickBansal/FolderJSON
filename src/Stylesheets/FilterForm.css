input[type="text"] {
    width: 350px;
    box-sizing: border-box;
    border: 2px solid black;
    border-radius: 10px;
    font-size: 20px;
    padding: 5px 10px;
    float: right;
    margin-bottom: 10px;
}

input[type="text"]:focus {
    outline: none;
}
