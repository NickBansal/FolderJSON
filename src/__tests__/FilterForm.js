import "../setupTest";
import React from "react";
import FilterForm from "../Components/FilterForm";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe.only("<FilterForm />", () => {
  let wrapper;
  const handleChange = jest.fn();
  beforeEach(() => {
    wrapper = shallow(<FilterForm handleChange={handleChange} />);
  });

  it("Should create a snapshot", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("Should call handleChange with value of the target value", () => {
    const inputField = wrapper.find("input");

    inputField.simulate("change", { target: { value: "A" } });
    expect(handleChange).toBeCalledWith("A");

    inputField.simulate("change", { target: { value: "ABC" } });
    expect(handleChange).toBeCalledWith("ABC");
  });
});
