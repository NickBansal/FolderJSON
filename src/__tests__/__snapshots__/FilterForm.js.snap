// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterForm /> Should create a snapshot 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="Filter..."
      type="text"
    />
  </form>
</div>
`;
